src/sia.js

```diff
@@ -0,0 +1,111 @@
+// sia.js: a lightweight node wrapper for starting, and communicating with
+// a Sia daemon (siad).
+import BigNumber from 'bignumber.js'
+import { spawn } from 'child_process'
+import request from 'request'
+
+// sia.js error constants
+export const errCouldNotConnect = new Error('could not connect to the Sia daemon')
+
+// Siacoin -> hastings unit conversion functions
+// These make conversion between units of Sia easy and consistent for developers.
+// Never return exponentials from BigNumber.toString, since they confuse the API
+BigNumber.config({ EXPONENTIAL_AT: 1e+9 })
+BigNumber.config({ DECIMAL_PLACES: 30 })
+
+const hastingsPerSiacoin = new BigNumber('10').toPower(24)
+const siacoinsToHastings = (siacoins) => new BigNumber(siacoins).times(hastingsPerSiacoin)
+const hastingsToSiacoins = (hastings) => new BigNumber(hastings).dividedBy(hastingsPerSiacoin)
+
+// Call makes a call to the Sia API at `address`, with the request options defined by `opts`.
+// returns a promise which resolves with the response if the request completes successfully
+// and rejects with the error if the request fails.
+const call = (address, opts) => new Promise((resolve, reject) => {
+	let callOptions = opts
+	if (typeof opts === 'string') {
+		callOptions = { url: opts }
+	}
+	callOptions.url = 'http://' + address + callOptions.url
+	callOptions.json = true
+	callOptions.headers = {
+		'User-Agent': 'Sia-Agent',
+	}
+
+	request(callOptions, (err, res, body) => {
+		if (!err && res.statusCode !== 200) {
+			reject(body)
+		} else if (!err) {
+			resolve(body)
+		} else {
+			reject(err)
+		}
+	})
+})
+
+// launch launches a new instance of siad using the flags defined by `settings`.
+// this function can `throw`, callers should catch errors.
+// callers should also handle the lifecycle of the spawned process.
+const launch = (path, settings) => {
+	const defaultSettings = {
+		'api-addr': 'localhost:9980',
+		'host-addr': ':9982',
+		'rpc-addr': ':9981',
+		'authenticate-api': false,
+		'disable-api-security': false,
+		'modules': 'cghmrtw',
+	}
+	const mergedSettings = Object.assign(defaultSettings, settings)
+	const filterFlags = (key) => mergedSettings[key] !== false
+	const mapFlags = (key) => {
+		if (typeof mergedSettings[key] === 'boolean') {
+			return '--' + key
+		} else {
+			return '--' + key + ' ' + mergedSettings[key]
```

Comment:
why the different handling for booleans? Can't you unify this with `'--' + key + '=' + mergedSettings[key]`?